src/include/miopen/readonlyramdb.hpp

```diff
@@ -106,6 +105,46 @@ class ReadonlyRamDb
     void Prefetch(const std::string& path, bool warn_if_unreadable);
 };
 
+struct FindRamDb : ReadonlyRamDb
+{
+    const std::unordered_map<std::string, std::string>& find_db_init(std::string arch_cu);
+    FindRamDb(std::string path, std::string _arch, std::size_t _num_cu)
+        : ReadonlyRamDb(":memory:" + path, _arch, _num_cu)
+    {
+        static const auto m = find_db_init(arch + "_" + std::to_string(num_cu));
+        cache               = m;
+    }
+    // Override GetCached, since FindRamDb does not have state or init overhead
+    static FindRamDb& GetCached(const std::string& path,
+                                bool /*warn_if_unreadble*/,
+                                const std::string& _arch,
+                                const std::size_t _num_cu)
+    {
+        static auto inst = new FindRamDb{path, _arch, _num_cu};
```

Comment:
You may just use an instance here instead of a pointer.

Reply:
Fixed